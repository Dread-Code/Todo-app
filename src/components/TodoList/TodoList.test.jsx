import React from "react";
import { shallow } from "enzyme";
import { TodoList } from "./TodoList";

describe("TodoList", () => {
  const handleToggle = jest.fn();
  const deleteTodo = jest.fn();
  const todos = [
    {
      id: 1625964511944,
      desc: "Estudiar jest",
      done: false,
    },
    {
      id: 1625964511945,
      desc: "Estudiar Mongo",
      done: false,
    },
  ];
  const wrapper = shallow(
    <TodoList
      todos={todos}
      handleToggle={handleToggle}
      deleteTodo={deleteTodo}
    />
  );
  test("should match snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have 2 TodoItem", () => {
    expect(wrapper.find("TodoItem").length).toBe(todos.length);
  });

  test("should have rigth props", () => {
    expect(wrapper.find("TodoItem").at(0).prop("deleteTodo")).toEqual(
      expect.any(Function)
    );
    expect(wrapper.find("TodoItem").at(0).prop("handleToggle")).toEqual(
      expect.any(Function)
    );
  });
});
