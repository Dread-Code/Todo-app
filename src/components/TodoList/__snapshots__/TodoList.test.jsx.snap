// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should match snapshoot 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoItem
      deleteTodo={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="1625964511944"
      todo={
        Object {
          "desc": "Estudiar jest",
          "done": false,
          "id": 1625964511944,
        }
      }
    />
    <TodoItem
      deleteTodo={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="1625964511945"
      todo={
        Object {
          "desc": "Estudiar Mongo",
          "done": false,
          "id": 1625964511945,
        }
      }
    />
  </ul>
</Fragment>
`;
